cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/updatetemplates/UpdateTemplatesDialog.java

```diff
@@ -0,0 +1,132 @@
+            sc.init(null, trustAllCerts, new java.security.SecureRandom());
+            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+        } catch (Exception e) {
+        }
+
+    }
+
+    @Override
+    protected Control createDialogArea(Composite parent) {
+        MDC.put(InfrastructureConstants.CORRELATION_ID, UUID.randomUUID().toString());
+
+        getShell().setText(UpdateTemplateDialogs.DIALOG_TITLE);
+        Composite contentParent = new Composite(parent, SWT.NONE);
+        GridData gridData = new GridData(GridData.FILL, GridData.FILL, true, true);
+        contentParent.setLayoutData(gridData);
+
+        GridLayout layout = new GridLayout(2, false);
+        layout.marginWidth = 15;
+        layout.marginHeight = 15;
+        contentParent.setLayout(layout);
+
+        Label introduction = new Label(contentParent, SWT.LEFT);
+        gridData = new GridData(GridData.FILL, GridData.FILL, true, true);
+        gridData.widthHint = 450;
+        gridData.horizontalSpan = 2;
+        introduction.setText("The following template folders will be downloaded from maven central repository:");
+        introduction.setLayoutData(gridData);
+
+        GridData leftGridData = new GridData(GridData.BEGINNING, GridData.CENTER, true, false);
+        leftGridData.widthHint = 320;
+        GridData rightGridData = new GridData(GridData.CENTER, GridData.CENTER, false, false);
+        rightGridData.widthHint = 80;
+        Label label = new Label(contentParent, SWT.NONE);
+        label.setText("templates-oasp4j");
+        label.setLayoutData(leftGridData);
+        MDC.remove(InfrastructureConstants.CORRELATION_ID);
+        return contentParent;
+    }
+
+    @Override
+    protected void createButtonsForButtonBar(Composite parent) {
+
+        Button button = createButton(parent, IDialogConstants.OK_ID, "Download", false);
+        button.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                try {
+                    ResourcesPluginUtil.downloadJar(false);
+                    MessageDialog dialog = new MessageDialog(Display.getDefault().getActiveShell(), "Information!!",
+                        null, "Downloaded succesfully", MessageDialog.INFORMATION, new String[] { "Ok" }, 1);
+                    dialog.setBlockOnOpen(true);
+                    dialog.open();
+                } catch (MalformedURLException e1) {
+                    LOG.error("An exception occurred  while downloading Jars from Maven Central", e1);
+                    throw new CobiGenRuntimeException("Invalid maven centrol repo url or path doesn't exist " + e1);
+                } catch (IOException e1) {
+                    LOG.error("An exception occurred  while while reading or writing Jar at .metadata folder", e1);
+                    throw new CobiGenRuntimeException("Failed while reading or writing Jar at .metadata folder" + e1);
```

Comment:
1. correct to "Failed reading or writing Jar at "+ CONSTANT REF +" folder."
2. log or throw. Never do both. In this case, throw.

Reply:
ok, I saw some code having both. I will do it.